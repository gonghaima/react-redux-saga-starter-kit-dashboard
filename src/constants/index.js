import keyMirror from "fbjs/lib/keyMirror";

const ActionTypes = keyMirror({
  EXCEPTION: undefined,
  GET_PRODUCTS: undefined,
  PRODUCT_GET_SUCCESS: undefined,
  PRODUCT_GET_FAILURE: undefined,
  SELECTION_CHANGE: undefined,
  PAGE_CHANGE: undefined,
  GET_USERS: undefined,
  USER_GET_SUCCESS: undefined,
  USER_GET_FAILURE: undefined,
  FILTER_USERS: undefined
});

const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error"
};

export { ActionTypes, STATUS };
