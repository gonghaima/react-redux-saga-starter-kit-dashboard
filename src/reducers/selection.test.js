import reducer from './selection';
import { ActionTypes } from '../constants';

describe('Selection reducer', () => {
  it('should return the initial state', () => {
    expect(reducer.selection(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.SELECTION_CHANGE}`, () => {
    expect(
      reducer.selection(undefined, {
        type: ActionTypes.SELECTION_CHANGE,
        payload: { itemPerPage: [2, 4, 10] }
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PAGE_CHANGE}`, () => {
    expect(
      reducer.selection(undefined, {
        type: ActionTypes.PAGE_CHANGE,
        payload: {}
      })
    ).toMatchSnapshot();
  });
});
