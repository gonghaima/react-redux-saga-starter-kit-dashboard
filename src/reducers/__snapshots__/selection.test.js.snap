// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection reducer should handle PAGE_CHANGE 1`] = `
Object {
  "currentPage": undefined,
  "itemPerPage": 4,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;

exports[`Selection reducer should handle SELECTION_CHANGE 1`] = `
Object {
  "currentPage": 0,
  "itemPerPage": undefined,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;

exports[`Selection reducer should return the initial state 1`] = `
Object {
  "currentPage": 0,
  "itemPerPage": 4,
  "selectionItems": Array [
    4,
    8,
    10,
    20,
    40,
  ],
}
`;
